### PR TITLE
`discuss-button`: new option - compact account drop down

### PR DESCRIPTION
**Pull request by retronbv**
Resolves None

### Changes

Adds a compact navbar addon that removes the username from the navbar and adds it in the dropdown.
image
image

### Reason for changes

Makes the Scratch navbar UI look cleaner.

### Tests

I have tested in scratch-www and in scratchr2. 

### Issues

With the `scratchr2` addon there is a slight margin error (see image below).

scratchr2 margin error

There is also a problem with the `account-settings-capitalize` addon. I presume this could be fixed via checking if this addon is enabled with `addon.self.getEnabledAddons()` in the account settings capitalization addon.

I would appreciate it if someone would take the liberty to PR into my fork with fixes to the above issues.

### CODE REVIEW

File: addons/compact-nav/addon.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "Compact Navbar",
```
Comment (RedGuy12) [inline addons/compact-nav/addon.json:2]: This name is a bit broad for such a narrow addon

File: addons/compact-nav/addon.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  "description": "Makes a more compact navbar.",
```
Comment (RedGuy12) [inline addons/compact-nav/addon.json:3]: Elaborate?

**Review by RedGuy12 — CHANGES_REQUESTED** [review]
Also enable actions on your fork

**Comment by Chiroyce1** [comment]
The navbar is the whole part
image

So I think navbar drop-down or something similar is a better name.

File: addons/compact-nav/addon.json (hunk 2 above)
Comment (RIMOPA) [reply addons/compact-nav/addon.json:3]: Moves the username of the navbar pop-up to inside the pop-up, making a more compact navbar.

**Comment by retronbv** [comment]
> The navbar is the whole part
> 
> image
> 
> 
> 
> So I think navbar drop-down or something similar is a better name.

Yes, i agree. I am actually re implementing this into the customizable navbar addon.

**Comment by Samq64** [comment]
If this will be part of custom navbar then maybe an option to remove the duplicate my stuff button from the navbar could be useful.

**Comment by mxmou** [comment]
> There is also a problem with the `account-settings-capitalize` addon. I presume this could be fixed via checking if this addon is enabled with `addon.self.getEnabledAddons()` in the account settings capitalization addon.

You could avoid this by replacing the "Profile" text with the username instead of adding a new item.

**Comment by retronbv** [comment]
> > There is also a problem with the `account-settings-capitalize` addon. I presume this could be fixed via checking if this addon is enabled with `addon.self.getEnabledAddons()` in the account settings capitalization addon.
> 
> You could avoid this by replacing the "Profile" text with the username instead of adding a new item.

I feel people would ask where the profile button went after they enable this... Maybe `[USERNAME] (Profile)` would be better text. I will get working on this

**Comment by Norbiros** [comment]
> > > There is also a problem with the `account-settings-capitalize` addon. I presume this could be fixed via checking if this addon is enabled with `addon.self.getEnabledAddons()` in the account settings capitalization addon.
> > 
> > 
> > You could avoid this by replacing the "Profile" text with the username instead of adding a new item.
> 
> I feel people would ask where the profile button went after they enable this... Maybe `[USERNAME] (Profile)` would be better text. I will get working on this

What if user have very long nickname? `(Profile)` wouldn't be visible.

**Comment by mxmou** [comment]
> Why not say `<username>'s profile`?

It could be too long. It also wouldn't clearly indicate that it's the username of the user who is currently logged in.

**Comment by WorldLanguages** [comment]
> Why not say `<username>'s profile`?

This is also harder to localize into other languages.

**Comment by Secret-chest** [comment]
> > Why not say `<username>'s profile`?
> 
> This is also harder to localize into other languages.

We can use placeholders.

**Comment by Secret-chest** [comment]
> > Why not say `<username>'s profile`?
> 
> It could be too long. It also wouldn't clearly indicate that it's the username of the user who is currently logged in.

Have it as an option?

**Comment by apple502j** [comment]
@Secret-chest Doesn't seem worth it.

**Comment by Secret-chest** [comment]
> @Secret-chest Doesn't seem worth it.

Seriously? Are you GNOME?

**Comment by mxmou** [comment]
> > > Why not say `<username>'s profile`?
> > 
> > 
> > This is also harder to localize into other languages.
> 
> We can use placeholders.

Possessive forms aren't as simple as adding `'s` in every language. For example, the correct form might depend on the user's gender. This means that it would have to be translated as "profile of user &lt;username&gt;" or something similar, which would be unnecessarily long.

> > It could be too long. It also wouldn't clearly indicate that it's the username of the user who is currently logged in.
> 
> Have it as an option?

It wouldn't be very useful because it doesn't change the addon's functionality in any way.

**Review by WorldLanguages — DISMISSED** [review]
Needs Prettier lint before merging

**Comment by WorldLanguages** [comment]
> You could avoid this by replacing the "Profile" text with the username instead of adding a new item.

@mxmou Do you want to implement this?

**Comment by retronbv** [comment]
If mxmou could that would be great, the way I tried was very sucky and didn't look good.

**Comment by retronbv** [comment]
ill run prettier

**Review by TheColaber — APPROVED** [review]

**Comment by Secret-chest** [comment]
Having a multiline item here is very strange. Why just not use Profile (username)?

File: addons/discuss-button/addon.json [hunk 3]
```diff
@@ -59,13 +75,19 @@
...
+      "type": "boolean",
+      "default": false
     }
   ],
   "versionAdded": "1.0.0",
   "latestUpdate": {
-    "version": "1.24.0",
-    "newSettings": ["items"],
-    "isMajor": true
+    "version": "1.26.0",
```
Comment (apple502j) [inline addons/discuss-button/addon.json:88]:
```suggestion
    "version": "1.28.0",
```

**Comment by retronbv** [comment]
pog

**Pull request merged**